src/migration/1715592299433-SourceFlagFeaturedIndex.ts

```diff
@@ -0,0 +1,17 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class SourceFlagFeaturedIndex1715592299433
+  implements MigrationInterface
+{
+  name = 'SourceFlagFeaturedIndex1715592299433';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `CREATE INDEX "IDX_source_flags_featured" ON source USING HASH (((flags->'featured')::boolean))`,
```

Comment:
We might have to check since we updated postgres not sure we need the hash and casting @capJavert ?

Reply:
HASH is generally more efficient (storage wise) for equality operations, thats why we use it for this bool flags. 

As for casting we can check with `EXPLAIN ANALYZE` to see if it is hit without cast, although it might also be depending on type of equality check we do, look at my comment here https://github.com/dailydotdev/daily-api/pull/1902#discussion_r1598250440, since we are gonna cast to boolean either way.

Reply:
Yeah nothing major, just remembered we only did certain things because the previous version didn't support everything right? (but yeah if we already cast it's fine)